crates/ruff_python_formatter/src/statement/stmt_try.rs

```diff
@@ -1,13 +1,116 @@
+    > where Self: 'a;
+
+    fn format(&self) -> Self::Format<'_> {
+        FormatRefWithRule::new(self, FormatExceptHandler::default())
+    }
+}
+
 impl FormatNodeRule<StmtTry> for FormatStmtTry {
     fn fmt_fields(&self, item: &StmtTry, f: &mut PyFormatter) -> FormatResult<()> {
-        write!(f, [not_yet_implemented(item)])
+        let StmtTry {
+            range: _,
+            body,
+            handlers,
+            orelse,
+            finalbody,
+        } = item;
+
+        let comments_info = f.context().comments().clone();
+        let mut dangling_comments = comments_info.dangling_comments(item.as_any_node_ref());
+
+        write!(
+            f,
+            [
+                text("try:"),
+                hard_line_break(),
+                block_indent(&body.format()),
+            ]
+        )?;
+
+        let mut last_node = body.last();
+
+        for handler in handlers {
+            let handler_comments_start =
+                dangling_comments.partition_point(|comment| comment.slice().end() <= handler.end());
+            let (handler_comments, rest) = dangling_comments.split_at(handler_comments_start);
+            dangling_comments = rest;
+
+            write!(
+                f,
+                [
+                    leading_alternate_branch_comments(handler_comments, last_node),
+                    &handler.format()
+                ]
+            )?;
+            last_node = match handler {
+                ExceptHandler::ExceptHandler(handler) => handler.body.last(),
+            };
+        }
+
+        if let [.., last] = &orelse[..] {
+            let orelse_comments_start =
+                dangling_comments.partition_point(|comment| comment.slice().end() <= last.end());
+            let (orelse_comments, rest) = dangling_comments.split_at(orelse_comments_start);
+            dangling_comments = rest;
+            write!(
+                f,
+                [
+                    leading_alternate_branch_comments(orelse_comments, last_node),
+                    text("else:"),
+                    block_indent(&orelse.format())
+                ]
+            )?;
+            last_node = orelse.last();
+        }
+        if let [.., last] = &finalbody[..] {
+            let finally_comments_start =
+                dangling_comments.partition_point(|comment| comment.slice().end() <= last.end());
+            let (finally_comments, rest) = dangling_comments.split_at(finally_comments_start);
+            dangling_comments = rest;
+            write!(
+                f,
+                [
+                    leading_alternate_branch_comments(finally_comments, last_node),
+                    text("finally:"),
+                    block_indent(&finalbody.format())
+                ]
+            )?;
+        }
+        write!(f, [comments::dangling_comments(dangling_comments)])
     }
 
     fn fmt_dangling_comments(&self, _node: &StmtTry, _f: &mut PyFormatter) -> FormatResult<()> {
```

Comment:
GitHub doesn't allow me to comment on the right line. We can remove the TODo below and instead add a comment that the dangling comments are formatted as part of `fmt_fields`.